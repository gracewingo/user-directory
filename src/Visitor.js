import React from 'react';

const Visitor = ({user, showUserProfile }) => (
	<li
		className="visitor"
		key={user.id}
		data-id={user.id}
		onClick={() => showUserProfile(user.id)}
	>
		{user.first_name} {user.last_name}
	</li>
);

export default Visitor;
