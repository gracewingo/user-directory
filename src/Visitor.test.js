import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import sinon from 'sinon';
import { create } from 'react-test-renderer';

import Visitor from './Visitor';

configure({ adapter: new Adapter() });

const user = [ { id: 21, name: 'Bob' } ];
const showUserProfile = sinon.spy();


describe('Visitor Component', () => {
	it('matches the snapshot', () => {
		const visitor = create(<Visitor user={user} />);
		expect(visitor.toJSON()).toMatchSnapshot();
	});

	it('shallow render', () => {
		shallow(<Visitor user={user} />);
	});

	it('simulates click events', () => {
		const wrapper = shallow(<Visitor user={user} showUserProfile={showUserProfile} />);
		wrapper.find('li').simulate('click');
	});
});
