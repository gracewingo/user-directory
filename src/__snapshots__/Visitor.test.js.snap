// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visitor Component matches the snapshot 1`] = `
<li
  className="visitor"
  onClick={[Function]}
>
   
</li>
`;

exports[`Visitorcomponent Matches the snapshot 1`] = `
<li
  className="visitor"
  onClick={[Function]}
>
   
</li>
`;

exports[`Visitorcomponent matches the snapshot 1`] = `
<li
  className="visitor"
  onClick={[Function]}
>
   
</li>
`;
